test/stats.bats

```diff
@@ -39,3 +39,48 @@ function teardown() {
     echo $JSON | jq -e '.stats[0].memory.workingSetBytes.value > 0'
     [ "$status" -eq 0 ]
 }
+
+@test "container stats" {
+    # given
+    container2config=$(cat "$TESTDATA"/container_redis.json | python -c 'import json,sys;obj=json.load(sys.stdin);obj["name"] = ["podsandbox1-redis2"];obj["metadata"]["name"] = "podsandbox1-redis2"; json.dump(obj, sys.stdout)')
+    echo "$container2config" > "$TESTDIR"/container_redis2.json
+    run crictl runp "$TESTDATA"/sandbox_config.json
+    echo "$output"
+    [ "$status" -eq 0 ]
+    pod_id="$output"
+    run crictl create "$pod_id" "$TESTDATA"/container_redis.json "$TESTDATA"/sandbox_config.json
+    echo "$output"
+    [ "$status" -eq 0 ]
+    ctr1_id="$output"
+    run crictl create "$pod_id" "$TESTDIR"/container_redis2.json "$TESTDATA"/sandbox_config.json
+    echo "$output"
+    [ "$status" -eq 0 ]
+    ctr2_id="$output"
+    run crictl start "$ctr1_id"
+    echo "$output"
+    [ "$status" -eq 0 ]
+    run crictl start "$ctr2_id"
+    echo "$output"
+    [ "$status" -eq 0 ]
+
+    # when
+    run crictl stats -o json "$ctr1_id"
+    echo "$output"
+    [ "$status" -eq 0 ]
+    ctr1_stats_JSON="$output"
+
+    run crictl stats -o json "$crt2_id"
+    echo "$output"
+    [ "$status" -eq 0 ]
+    ctr2_stats_JSON="$output"
+
+    echo $ctr1_stats_JSON | jq -e '.stats[0].memory.workingSetBytes.value'
+    [ "$status" -eq 0 ]
```

Comment:
Not sure what you check here -- `$status` is set by bats' `run` function.

Reply:
if the json didn't have the value field then the jq command would fail

Reply:
I understand the `jq` part, but the variable being checked, `$status`, was set last time the `run ...` was executed, and it was already checked earlier. In other words, the value of `$status` has nothing to do with `echo .. | jq` invocation in the previous line.

Also, bats sets `-e -o pipefail` options for bash (or so I hope), so if `jq` fails the test fails, there is no need to specifically check its exit code.

So, this code has (1) no effect and (2) is misleading as the reader thinks we're checking `echo | jq` exit code here, but it's not true.

Reply:
oh good catch, I didn't notice there was no run

Reply:
It's a little trickier with the `bats` -- you have to remember this is not just `bash`.

Reply:
fixed!

Reply:
So, afaik bats set `set -e` and `set -o pipefail`, meaning that if any command in a pipeline `echo ... | jq ... ` fails, the test case fails.

In other words, the following two code blocks are roughly equivalent:

```bash
echo $ctr1_stats_JSON | jq -e '.stats[0].memory.workingSetBytes.value'
```

```bash
run echo $ctr1_stats_JSON | jq -e '.stats[0].memory.workingSetBytes.value'
[ "$status" -eq 0 ]
```

and unless you want to do something about command exit code (say check if it's 32) and/or output, there is no need to use run. Sorry I should have been clear in my earlier comment -- what I hinted at is that `[ "$status" -eq 0 ]` should be removed (rather than `run ` prefix added).

Can surely be addressed as a followup

Reply:
Here's an example:

```console
[kir@kir-rhat runc]$ cat a.bats 
@test "use run" {
	run ls -l | false
	[ "$status" -eq 0 ]
}

@test "don't use run" {
	ls -l | false
}
[kir@kir-rhat runc]$ bats a.bats 
 ✗ use run
   (in test file a.bats, line 2)
     `run ls -l | false' failed
 ✗ don't use run
   (in test file a.bats, line 7)
     `ls -l | false' failed

2 tests, 2 failures
```